### PR TITLE
Add collation parameter to the New-AzSqlInstance command

### PR DESCRIPTION
**Pull request by NMijat1024**
## Description

This PR adds possibility for user to set custom collations during MI create.

This option is already available in 'AzureRM.Sql-preview' branch. (https://github.com/Azure/azure-powershell/pull/7923)

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql/help/New-AzSqlInstance.md [hunk 1]
```diff
@@ -122,6 +122,19 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -Collation
+The collation of the Azure SQL Managed Instance to use.```yaml
```
Comment (petrajkogit) [inline src/Sql/Sql/help/New-AzSqlInstance.md]: Please move yaml tag into empty line

File: src/Sql/Sql/Sql.csproj [hunk 2]
```diff
@@ -10,8 +10,13 @@
     <RootNamespace>$(LegacyAssemblyPrefix)$(PsModuleName)</RootNamespace>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (petrajkogit) [inline src/Sql/Sql/Sql.csproj]: Can you remove this generated change?

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 3]
```diff
@@ -139,6 +139,15 @@ public class NewAzureSqlManagedInstance : ManagedInstanceCmdletBase
...
         public string ComputeGeneration { get; set; }
 
+        /// <summary>
+        /// Gets or sets the instance collation
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The collation of the Azure SQL Managed Instance to use.")]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("SQL_Latin1_General_CP1_CI_AS", "Latin1_General_100_CS_AS_SC")]
+        public string Collation { get; set; }
```
Comment (yaakoviyun) [inline src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: can't you define an enum of possible values?

File: src/Sql/Sql.Test/Sql.Test.csproj [hunk 4]
```diff
@@ -11,7 +11,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.22.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.24.0-preview" />
```
Comment (yaakoviyun) [inline src/Sql/Sql.Test/Sql.Test.csproj]: please update the new version in the sql.csproj too

Comment (NMijat1024) [reply src/Sql/Sql.Test/Sql.Test.csproj]: It is already updated

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs (hunk 3 above)
Comment (NMijat1024) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: Done.

**Pull request merged**